### PR TITLE
Changes for no control workflows

### PR DESCRIPTION
**Pull request by dankwart-de**
Make workflow more robust and easier to use regarding no control

### CODE REVIEW

File: src/de/dkfz/b080/co/common/WorkflowUsingMergedBams.groovy [hunk 1]
```diff
@@ -1,144 +1,173 @@
...
-            if (bamTumorMerged == null)
-                context.addErrorEntry(ExecutionContextError.EXECUTION_NOINPUTDATA.expand("Tumor bam is missing"));
-            return false;
+        if (!initialBamFiles) {
+            context.addErrorEntry(ExecutionContextError.EXECUTION_NOINPUTDATA.expand("Did not find any bam files."))
+            return false
+        }
+        if (isNoControlWorkflow) {
+            boolean foundAll = true
+            initialBamFiles.each { BasicBamFile it -> foundAll &= it && ((COFileStageSettings) it.getFileStage()).sample.sampleType == Sample.SampleType.TUMOR }
```
Comment (vinjana) [inline src/de/dkfz/b080/co/common/WorkflowUsingMergedBams.groovy]: `boolean foundAll = collect { bla }.all()`

File: src/de/dkfz/b080/co/common/WorkflowUsingMergedBams.groovy [hunk 2]
```diff
@@ -1,144 +1,173 @@
...
+                context.addErrorEntry(ExecutionContextError.EXECUTION_NOINPUTDATA.expand("Not all found files are tumor bam files."))
+                return false
+            }
+
+        } else {
+            boolean foundAll = true
+            BasicBamFile bamControlMerged = initialBamFiles[0]
+
+            for (int i = 1; i < initialBamFiles.size(); i++) {
+                if (initialBamFiles[i] == null || !((COFileStageSettings) initialBamFiles[i].getFileStage()).sample.sampleType == Sample.SampleType.TUMOR) {
```
Comment (vinjana) [inline src/de/dkfz/b080/co/common/WorkflowUsingMergedBams.groovy]: Maybe do `initialBamFiles.collect { it != null }.all` at the beginning, to get rid of the if's down here.

File: src/de/dkfz/b080/co/common/WorkflowUsingMergedBams.groovy [hunk 3]
```diff
@@ -1,144 +1,173 @@
...
-            return false;
+            return false
 
         //TODO Low priority. There were thoughts to have workflows which support multi-tumor samples, this it not supported by any workflow now.
         if (context.getConfiguration().getConfigurationValues().getBoolean(WORKFLOW_SUPPORTS_MULTI_TUMOR_SAMPLES, false)) {
-            return executeMulti(context, initialBamFiles);
+            return executeMulti(context, initialBamFiles)
         }
-        return execute(context, initialBamFiles[0], initialBamFiles[1]);
+        boolean isNoControlWorkflow = getflag(context, IS_NO_CONTROL_WORKFLOW, false)
```
Comment (vinjana) [inline src/de/dkfz/b080/co/common/WorkflowUsingMergedBams.groovy]: extract query into method `getIsNoControlWorkflow`?

File: resources/configurationFiles/commonCOWorkflowsSettings.xml [hunk 4]
```diff
@@ -30,16 +36,6 @@
...
-        <tool name='checkBwaAndIndexVersions' value='checkBWAAndIndexVersions.sh' basepath='tools'/>
-        <tool name="annotateVcfFile" value="annotate_vcf.pl" basepath="tools"/>
-        <tool name="convertNewcolsToVcf" value="newCols2vcf.pl" basepath="tools"/>
-        <tool name="createpipes" value="createpipes.pl" basepath="tools"/>
-        <tool name="headeredFileConcatenator" value="headeredFileConcatenator.pl" basepath="tools"/>
-        <tool name="newColsToVcf" value="newCols2vcf.pl" basepath="tools"/>
-        <tool name="processAnnovar" value="processAnnovarOutput.pl" basepath="tools"/>
-        <tool name="seqContextAnnotator" value="seqContext_annotator.pl" basepath="tools"/>
-        <tool name="vcfToAnnovar" value="vcf_to_annovar.pl" basepath="tools"/>
-        <tool name="bamIsComplete" value="bamIsComplete.sh" basepath="tools"/>
```
[the remark this answers is not in the document]
Comment (dankwart-de) [reply resources/configurationFiles/commonCOWorkflowsSettings.xml]: Only bamIsComplete right? I removed the others. Kept all the basic bam tools

[the remark this answers is not in the document]
Comment (vinjana) [reply resources/configurationFiles/commonCOWorkflowsSettings.xml]: Right, bamIsComplete was just added. About the other scripts I cannot tell much. You also had the idea of putting the scripts into an extra plugin.

**Pull request closed without merging**